### PR TITLE
add support for n dimensions bijector

### LINKED ISSUE #23: Smooth Normalizing Flow N dimensions
Finally works for more than 2d :) 
I changed the bijector and the NN in the NF and tested it on 6d: [notebook](https://colab.research.google.com/drive/1YSjUofeeAhCqxb4L2tkiyInZFCxD-x27?usp=sharing)

### PR DESCRIPTION
**Pull request by Justinezgh**
This small PR solved #23 [notebook](https://colab.research.google.com/drive/1YSjUofeeAhCqxb4L2tkiyInZFCxD-x27?usp=sharing)

### CODE REVIEW

File: tests/test_bijectors.py [hunk 1]
```diff
@@ -92,3 +93,116 @@ def test_ramp_bijector_inverse_exp():
 
 
 
+def _make_inv_forward_MixtureAffineSigmoidBijector(t, a, b, c, nb_dimension, nb_component):
```
Comment (EiffL) [inline tests/test_bijectors.py:96]: hummmm I don;t think it is necessary here to create a separate function. You are using a different function for each test, so it might as well be directly the test function itself. Otherwise this is a bit cluttered.

Here is how I would write this function:
```python
def test_bijectorMixtureAffineSigmoidBijector_inverse():
  """
  Testing that the inverse transform of MixtureAffineSigmoidBijector is indeed the inverse of 
  forward.
  """
  batch_size = 100

  for params in _test_params_MASB:
    x = jax.random.uniform(jax.random.PRNGKey(0), shape=[batch_size,params['nb_dimension']])
    
    # shape for components of the transform
    shape = [batch_size, params['nb_dimension'], params['nb_component']]
    bij = MixtureAffineSigmoidBijector(
                              jnp.ones(shape)*a,
                              jnp.ones(shape)*b,
                              jnp.ones(shape)*c,
                              jax.nn.softmax(jax.random.uniform(jax.random.PRNGKey(1), shape=shape)))

    assert_allclose(bij.inverse(bij.forward(x)*1.), x, rtol=2e-2, atol=1e-5)
```
It's just one function instead of 2, a lot more compact :-)

File: tests/test_bijectors.py [hunk 2]
```diff
@@ -92,3 +93,116 @@ def test_ramp_bijector_inverse_exp():
...
+
+def test_bijectorMixtureAffineSigmoidBijector_forward_dim():
+  """
+  Testing forward dimension of MixtureAffineSigmoidBijector
+  """
+  batch_size = 100
+
+  for params in _test_params_inv_MASB:
+    x = jax.random.uniform(jax.random.PRNGKey(0), shape=[batch_size,params['nb_dimension']])
+    assert_allclose(_make_forward_MixtureAffineSigmoidBijector(x, **params).shape, (batch_size,params['nb_dimension']) , rtol=2e-3, atol=1e-6)
```
Comment (EiffL) [inline tests/test_bijectors.py:179]: for shapes, which are integers, it's better to use directly `assert_equal` because you don't have issues of numerical accuracy.

**Review by EiffL — CHANGES_REQUESTED** [review]
Made a few optimization comments on the tests but looks good otherwise  :-)

**Comment by EiffL** [comment]
@Justinezgh ? do you agree with my comments?

**Review by EiffL — APPROVED** [review]
LGTM :-)
